src/window/basic_frame.rs

```diff
@@ -429,123 +433,160 @@ impl Frame for BasicFrame {
             }
 
             // Create the buffers
-            // -> top-subsurface
+            // -> head-subsurface
             let buffer = pool.buffer(
                 0,
-                (width + 2 * DECORATION_SIZE) as i32,
-                DECORATION_TOP_SIZE as i32,
-                4 * (width + 2 * DECORATION_SIZE) as i32,
+                width as i32,
+                HEADER_SIZE as i32,
+                4 * width as i32,
                 wl_shm::Format::Argb8888,
             ).implement(|_, _| {});
-            self.inner.parts[TOP]
+            self.inner.parts[HEAD]
                 .subsurface
-                .set_position(-(DECORATION_SIZE as i32), -(DECORATION_TOP_SIZE as i32));
+                .set_position(0, -(HEADER_SIZE as i32));
+            self.inner.parts[HEAD].surface.attach(Some(&buffer), 0, 0);
+            if self.surface_version >= 4 {
+                self.inner.parts[HEAD].surface.damage_buffer(
+                    0,
+                    0,
+                    (width + 2 * BORDER_SIZE) as i32,
+                    HEADER_SIZE as i32,
+                );
+            } else {
+                // surface is old and does not support damage_buffer, so we damage
+                // in surface coordinates and hope it is not rescaled
+                self.inner.parts[HEAD].surface.damage(
+                    0,
+                    0,
+                    (width + 2 * BORDER_SIZE) as i32,
```

Comment:
same here, only need to damage `width`

Reply:
I also updated the damage to the side buffers as I changed their height to extend for the full height of the window